_commands/tools/make.md

```diff
@@ -2,8 +2,115 @@
+# Here is a comment!
+target: dependencies
+	[tab] command
+~~~
+
+If you just type `make` without specifying a target, it will run the first target in the file. If you type a specific target name like `make test` it will run the target with that name. If we haven't run the target before or if any of the dependencies have changed, make will go ahead and run whatever command is specified after the tab character on the next line.
+
+#### Example command
+~~~ bash
+$ make Matrix_tests.exe
+g++ --std=c++11 -Wall -Werror -pedantic -O1 Matrix_public_test.cpp Matrix.cpp Matrix_test_helpers.cpp -o Matrix_public_test.exe
+~~~
+
+##### Break it down
+`make Matrix_tests.exe` refers to the target by the same name. This target stitches together all the cpp files needed to compile `Matrix_public_test.exe`.
+
+#### Example command
+~~~ bash
+$ make clean
+rm -vf *.exe *.out.txt *.out.ppm
+Image_public_test.exe
+Image_tests.exe
+Matrix_public_test.exe
+Matrix_tests.exe
+processing_public_tests.exe
+resize.exe
+~~~
+
+##### Break it down
+`make clean` in this Makefile is set up to remove all executables, .out.txtm and .out.ppm files. This is good to have if you want a nice clean start before you compile again.
+
+
+### Final Remarks
+
+Makefiles can save you a lot of time from having to type out long compilation sequences or running test suites!
+
+
+~~~ bash
+# Makefile
+# Build rules for EECS 280 project 2
+
+# Compiler
+CXX = g++
+
+# Set this at the command line using: make debug_or_optimize=<flag> <target> ...
+debug_or_optimize = -O1
+
+# Compiler flags
+CXXFLAGS = --std=c++11 -Wall -Werror -pedantic $(debug_or_optimize)
+
+test: release
+	./Matrix_public_test.exe
+	./Image_public_test.exe
+	./processing_public_tests.exe dog
+	./processing_public_tests.exe crabster
+	./processing_public_tests.exe horses
+	./resize.exe dog.ppm dog_4x5.out.ppm 4 5
+	diff dog_4x5.out.ppm dog_4x5.correct.ppm
+
+release: Matrix_public_test.exe Matrix_tests.exe Image_public_test.exe \
+		 Image_tests.exe processing_public_tests.exe resize.exe
+
+debug: debug_or_optimize = -g
+debug: Matrix_public_test.exe Matrix_tests.exe Image_public_test.exe \
+	   Image_tests.exe processing_public_tests.exe resize.exe
+
+Matrix_public_test.exe: Matrix_public_test.cpp Matrix.cpp Matrix_test_helpers.cpp
+	$(CXX) $(CXXFLAGS) $^ -o $@
+
+Matrix_tests.exe: Matrix_tests.cpp Matrix.cpp Matrix_test_helpers.cpp
+	$(CXX) $(CXXFLAGS) $^ -o $@
+
+Image_public_test.exe: Image_public_test.cpp Matrix.cpp Image.cpp \
+			Matrix_test_helpers.cpp Image_test_helpers.cpp
+	$(CXX) $(CXXFLAGS) $^ -o $@
+
+Image_tests.exe: Image_tests.cpp Matrix.cpp Image.cpp Matrix_test_helpers.cpp \
+			Image_test_helpers.cpp
+	$(CXX) $(CXXFLAGS) $^ -o $@
+
+processing_public_tests.exe: processing_public_tests.cpp Matrix.cpp \
+				Image.cpp processing.cpp \
+				Matrix_test_helpers.cpp Image_test_helpers.cpp
+	$(CXX) $(CXXFLAGS) $^ -o $@
+
+resize.exe: resize.cpp Matrix.cpp Image.cpp processing.cpp
+	$(CXX) $(CXXFLAGS) $^ -o $@
```

Comment:
There are some special characters in this script that aren't very clear about what their doing. You don't have to describe them, but definitely link to this page.

https://www.gnu.org/software/make/manual/make.html#Automatic-Variables